src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -24,4 +24,109 @@
   protected HttpBatchSourceConfig(String referenceName) {
     super(referenceName);
   }
+
+  private HttpBatchSourceConfig(HttpBatchSourceConfigBuilder builder) {
+    super(builder.referenceName);
+    this.url = builder.url;
+    this.httpMethod = builder.httpMethod;
+    this.headers = builder.headers;
+    this.format = builder.format;
+    this.oauth2Enabled = builder.oauth2Enabled;
+    this.errorHandling = builder.errorHandling;
+    this.retryPolicy = builder.retryPolicy;
+    this.maxRetryDuration = builder.maxRetryDuration;
+    this.connectTimeout = builder.connectTimeout;
+    this.readTimeout = builder.readTimeout;
+    this.paginationType = builder.paginationType;
+    this.verifyHttps = builder.verifyHttps;
+  }
+
+  /**
+   * Builder for HttpBatchSourceConfig
+   */
+  public static class HttpBatchSourceConfigBuilder {
```

Comment:
Add a public static builder method in the config class to return new object of builder.
See this for reference, https://github.com/data-integrations/google-cloud/blob/6fa17791c6dfd10ec26a37564c4c618401206ee1/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java#L680

Reply:
done